exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/FunctionImplementationRegistry.java

```diff
@@ -160,7 +168,7 @@ public DrillFuncHolder findDrillFunction(FunctionResolver functionResolver, Func
     FunctionResolver exactResolver = FunctionResolverFactory.getExactResolver(functionCall);
     DrillFuncHolder holder = exactResolver.getBestMatch(functions, functionCall);
 
-    if (holder == null) {
+    if (holder == null && useDynamicUdfs) {
```

Comment:
Ah, now I see what's happening (I hope...) I pushed another commit that makes the suggested changes. 

I wonder, do we have any unit tests for the ambiguous-function case? The unit tests passed with both the original and this new version, so I wonder if we have a hole in our test coverage?